app/src/test/java/org/mozilla/fenix/utils/BrowsersCacheTest.kt

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+@file:Suppress("DEPRECATION")
+
+package org.mozilla.fenix.utils
+
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import android.content.pm.PackageInfo
+import android.content.pm.ResolveInfo
+import android.net.Uri
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.utils.Browsers
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(application = TestApplication::class)
+class BrowsersCacheTest {
+
+    // NB: There is always one more browser than pretendBrowsersAreInstalled installs because
+    // the application we are testing is recognized as a browser itself!
+
+    @Test
+    fun testInstallNewBrowsersWithoutReset() {
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        var browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
```

Comment:
nit: argument order should be reversed throughout the file. It's `assertEquals(expected, actual)`

Reply:
Very helpful! I completely meant to do that. Thank you!